packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -39,18 +39,13 @@ def _render_template(template_name: str, **template_args: Any) -> bytes:
     return template.render(**template_args)
 
 
-def _replace_enum_class_type(output: str) -> str:
-    pattern = "<enum '([^']+)'>"
-    return re.sub(pattern, r"\1", output)
-
-
 def _create_file(
     template_name: str, file_name: str, directory_out: pathlib.Path, **template_args: Any
 ) -> None:
     output_file = directory_out / file_name
 
     output = _render_template(template_name, **template_args).decode("utf-8")
-    output = _replace_enum_class_type(output)
+    output = replace_enum_class_type(output)
     formatted_output = black.format_str(
```

Comment:
I overlooked this in the previous reviews. Why are you replacing the type at this level? Doesn't the generator know when it's generating a type?

Reply:
Yes, the generator knows the enum type. The enum_type will be like this
`configuration_metadata = ParameterMetadata.Initialize(...,enum_type=Color)`

But when after rendering we see the output like this
```
self._configuration_metadata = {
    1: ParameterMetadata(
        display_name="Float In",
        type=2,
        repeated=False,
        default_value=0.05999999865889549,
        annotations={},
        message_type="",
        field_name="Float_In",
        enum_type=None,
    ),
    6: ParameterMetadata(
        display_name="Enum Array In",
        type=14,
        repeated=True,
        default_value=[3, 2],
        annotations={
            "ni/enum.values": '{"NONE": 0, "RED": 1, "GREEN": 2, "BLUE": 3}',
            "ni/type_specialization": "enum",
        },
        message_type="",
        field_name="Enum_Array_In",
        enum_type=<enum 'EnumInEnum'>,     # This representation instead of just Enum name.
    ),
}
```
Black formatting fails because of `enum_type=<enum 'EnumInEnum'>` where it has to be `enum_type=EnumInEnum`. So we are fixing this error here.

Reply:
Generating wrong output and then fixing it up later by running a regex over the whole output file is a big hack.

At the very least, reduce the scope of the hack by moving it to the code that generates the metadata: https://github.com/ni/measurement-plugin-python/blob/main/packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako#L93

A more robust fix would be to add code to the template to generate the metadata namedtuple, rather than relying on Python's str/repr implementation.

Reply:
I have updated by moving this conversion to `mako` template to update the `enum_type` of `configuration_metadata` and `output_metadata`.